src/types/st-array.ts

```diff
@@ -1,37 +1,83 @@
-import { makeClass } from "../utils/make-class";
-import { ensureArrayLikeIs, SerializedType } from "./serialized-type";
+import { SerializedTypeClass } from "./serialized-type";
 import { STObject } from "./st-object";
+import { BinaryParser } from "../serdes/binary-parser";
 
-const STArray = makeClass(
-  {
-    mixins: SerializedType,
-    inherits: Array,
-    statics: {
-      fromParser(parser) {
-        const array = new STArray();
-        while (!parser.end()) {
-          const field = parser.readField();
-          if (field.name === "ArrayEndMarker") {
-            break;
-          }
-          const outer = new STObject();
-          outer[field.name] = parser.readFieldValue(field);
-          array.push(outer);
-        }
-        return array;
-      },
-      from(value) {
-        return ensureArrayLikeIs(STArray, value).withChildren(STObject);
-      },
-    },
-    toJSON() {
-      return this.map((v) => v.toJSON());
-    },
-    toBytesSink(sink) {
-      this.forEach((so) => so.toBytesSink(sink));
-    },
-  },
-  undefined
-);
+const ARRAY_END_MARKER = 0xf1;
+const OBJECT_END_MARKER = 0xe1;
+
+/**
+ * Class for serializing and deserializing Arrays of Objects
+ */
+class STArray extends SerializedTypeClass {
```

Comment:
OOC: ST == "Serialized Type"?

Reply:
Yeah, ST is short for SerializedType, a convention taken from rippled (which may or may not still use it)